prov/psm/src/psmx_init.c

```diff
@@ -439,8 +439,16 @@ static int psmx_getinfo(uint32_t version, const char *node, const char *service,
 				goto err_out;
 			}
 
-			if (hints->domain_attr->mr_mode & FI_MR_BASIC)
+			if (hints->domain_attr->mr_mode == FI_MR_BASIC) {
 				mr_mode = FI_MR_BASIC;
+			} else if (hints->domain_attr->mr_mode == FI_MR_SCALABLE) {
+				mr_mode = FI_MR_SCALABLE;
+			} else if (hints->domain_attr->mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE)) {
+				FI_INFO(&psmx_prov, FI_LOG_CORE,
+					"hints->domain_attr->mr_mode has FI_MR_BASIC or FI_MR_SCALABLE "
+					"combined with other bits\n");
+				goto err_out;
+			}
```

Comment:
You may want to use ofi_check_mr_mode() to verify mr_mode.  The else if part here doesn't look right.  It only verifies that basic and scalable aren't set together.

Reply:
The else if part verifies that neither FI_MR_BASIC nor FI_MR_SCALABLE bit is set.

Reply:
actually - the check does look okay -- I understand the logic now

Reply:
All other bits are ignored and the default value is taken. The default value is 0 for 1.5 and FI_MR_SCALABLE for older versions.